packages/neos-ui-editors/src/SecondaryEditors/ImageCropper/index.js

```diff
@@ -136,11 +137,39 @@ export default class ImageCropper extends PureComponent {
         });
     }
 
+    handleCropComplete = (cropArea, cropAreaAbsolute) => {
+        const {onComplete, sourceImage, options} = this.props;
+        const {cropConfiguration} = this.state;
+        const currentAspectRatioStrategy = cropConfiguration.aspectRatioStrategy;
+        const pixelSnapping = $get('crop.aspectRatio.pixelSnapping', options);
+
+        if (pixelSnapping && currentAspectRatioStrategy && currentAspectRatioStrategy.width && currentAspectRatioStrategy.height) {
+            const imageWidth = $get('image.originalDimensions.width', sourceImage);
+            const imageHeight = $get('image.originalDimensions.height', sourceImage);
+
+            // normalize aspect ratio values by dividing by gcd
+            const calculateGcdRecursive = (a, b) => b ? calculateGcdRecursive(b, a % b) : a;
```

Comment:
This function should live outside of the component, because it does not depend on its internal state.

(For simplicity's sake, it can still live in this file, since it's not being used elsewhere)

Reply:
@patricekaufmann Do you have the time to move the function outside, like Wilhelm recommends?
We have feature freeze on Thursday 17th of November. It would be nice if we can make it to get this improvement of you for Neos 8.2, thank you  🙏

Reply:
@markusguenther As requested, I moved the function `calculateGcdRecursive` outside of the component. 

Furthermore I changed the rounding of the cropping area to always round down. Potentially rounding up could lead to an overflow of the cropping area and black borders on the generated image when the cropping area exceeded the image dimensions.